src/qibo/transpiler/pipeline.py

```diff
@@ -0,0 +1,221 @@
+    initial_layout: dict,
+    final_layout: dict,
+    native_gates: NativeType = NativeType.CZ,
+    check_circuit_equivalence=True,
+):
+    """Check that all transpiler passes have been executed correctly.
+
+    Args:
+        original_circuit (qibo.models.Circuit): circuit before transpiling.
+        transpiled_circuit (qibo.models.Circuit): circuit after transpiling.
+        connectivity (networkx.Graph): chip qubits connectivity.
+        initial_layout (dict): initial physical-logical qubit mapping.
+        final_layout (dict): final physical-logical qubit mapping.
+        native_gates: (NativeType): native gates supported by the hardware.
+    """
+    assert_connectivity(circuit=transpiled_circuit, connectivity=connectivity)
+    assert_decomposition(circuit=transpiled_circuit, two_qubit_natives=native_gates)
+    if original_circuit.nqubits != transpiled_circuit.nqubits:
+        qubit_matcher = Preprocessing(connectivity=connectivity)
+        original_circuit = qubit_matcher(circuit=original_circuit)
+    assert_placement(circuit=original_circuit, layout=initial_layout)
+    assert_placement(circuit=transpiled_circuit, layout=final_layout)
+    if check_circuit_equivalence:
+        assert_circuit_equivalence(
+            original_circuit=original_circuit,
+            transpiled_circuit=transpiled_circuit,
+            initial_map=initial_layout,
+            final_map=final_layout,
+        )
+
+
+class Passes:
+    """Define a transpiler pipeline consisting of smaller transpiler steps that are applied sequentially:
+
+    Args:
+        passes (list): list of passes to be applied sequentially,
+            if None default transpiler will be used, it requires hardware connectivity.
+        connectivity (nx.Graph): hardware qubit connectivity.
+    """
+
+    def __init__(
+        self,
+        passes: list = None,
+        connectivity: nx.Graph = None,
+        native_gates: NativeType = NativeType.CZ,
+    ):
+        self.native_gates = native_gates
+        if passes is None:
+            self.passes = self.default(connectivity)
+        else:
+            self.passes = passes
+        self.connectivity = connectivity
+
+    def default(self, connectivity: nx.Graph):
+        """Return the default transpiler pipeline for the required hardware connectivity."""
+        if not isinstance(connectivity, nx.Graph):
+            raise_error(
+                TranspilerPipelineError,
+                "Define the hardware chip connectivity to use default transpiler",
+            )
+        default_passes = []
+        # preprocessing
+        default_passes.append(Preprocessing(connectivity=connectivity))
+        # default placer pass
+        default_passes.append(Trivial(connectivity=connectivity))
+        # default_passes.append(ReverseTraversal(connectivity=connectivity, routing_algorithm=ShortestPaths(connectivity), depth=5))
+        # default router pass
+        default_passes.append(StarConnectivity())
```

Comment:
why is `StarConnectivity` used by default regardless of the `connectivity` passed?

Reply:
Because now sabre is not able to handle measurements and star connectivity is stable. when we will fix the issue with measurements we will mode to sabre as the default.